### DISCUSSION TITLE
Why not add functions "add reroute"for transfer node's output?

### DISCUSSION THREAD

**Discussion by se7enth** [Comfy-Org/ComfyUI#607] [Ideas]
There are reroute and node templates, that is very good design for any wokflow.But to connect and replace the existing working nodes somtimes is very troublesome, you need to connect them one by one on the original input node . Any workflow can use"reroute" node skip/swith to another, and I found there are some functions like "add lora","add clip skip" in model loader,"add save image""add hi-res fix“ in ksample, obviously, "add save image"or"add hi-res fix“ is a pre-defined working template for ksample ,but the "add reroute" is all node can output to, and connect any custom template.

![image](https://user-images.githubusercontent.com/434242/235781706-f74af65d-7c0b-46c7-b490-4b4bba59740f.png)
![image](https://user-images.githubusercontent.com/434242/235781968-7db50257-0c7d-4749-85cb-ee17d106db3f.png)

**Comment by se7enth** [comment]
Sorry, I found the function should be an attribute in right click menu  of any output slot. Now it there are "disconnect link" and "rename slot", Cloud "add reroute" become the third one?
let me explain the idea，for exsample,there is a workflow,
![image](https://user-images.githubusercontent.com/434242/235919105-84cc481e-5c23-432e-8146-1038b1363c9f.png)
I wanna chage VAE ,I had to add node and connect  nodes one by one , if there is ”add reroute“ in slot,We can easily transfer connections from existing nodes to a new VAE node via this route。

![image](https://user-images.githubusercontent.com/434242/235921547-9ca9b86b-75f4-4d2a-8418-b916e628ae68.png)
![image](https://user-images.githubusercontent.com/434242/235921697-859e6fa0-4d7a-4480-ad4b-99bc86c04d8f.png)
I think it would be useful，sometime we need change model，vae，prompt...and keep the original process in standby mode。The function would make that easy。

**Comment by morphles** [comment]
Was considering suggesting/trying to implement this myself. But yes, spawning reroutes is needed, could also be option on middle of line dot click.

**Comment by omar92** [comment]
i did that , on the suggestions

**Comment by coolzilj** [comment]
Are you suggesting something like this?

https://github.com/comfyanonymous/ComfyUI/assets/1059327/8f6bab43-6454-4b55-8e41-3c42ddb2c13c

https://x.com/Songzi39590361/status/1799730472362414151
